src/Docker.PowerShell/Cmdlets/GetContainerImage.cs

```diff
@@ -25,10 +37,17 @@ public class GetContainerImage : DkrCmdlet
         /// </summary>
         protected override async Task ProcessRecordAsync()
         {
-            foreach (var img in await DkrClient.Images.ListImagesAsync(
-                new ImagesListParameters() { All = All.ToBool() }))
+            var listParams = new ImagesListParameters() { All = (All || Id != null) };
+
+            foreach (var img in await DkrClient.Images.ListImagesAsync(listParams))
             {
-                WriteObject(img);
+                if (Id == null ||
+                    Id.Any(i => img.ID.Equals("sha256:" + i)) ||
+                    Id.Any(i => img.RepoTags.Any(r => r.StartsWith(i))) ||
+                    Id.Any(i => img.ID.StartsWith(i) || img.ID.StartsWith("sha256:" + i)))
```

Comment:
The one in the object always will, but this is my stupid way for accounting for the case where the user typed "sha256:" into the passed in ID.  Unlikely someone would do that by hand, but it may be in there if they are using a cached value that came out of an object at one point or something like that.